301/sort/challenges-03.test.js

```diff
@@ -0,0 +1,287 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1
+
+Write a function named sortBackwards that takes in an array of numbers and returns the same array, with the numbers sorted, highest to smallest.
+------------------------------------------------------------------------------------------------ */
+
+const sortBackwards = (arr) => {
+arr.sort(function(a, b){return b-a});
+return arr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named alphabetize that takes in an array of strings and returns the same array with the strings sorted alphabetically.
+
+In this alphabetization, capital letters come before lower case letters.
+
+For example, ['Alphabet', 'Zebra', 'alphabet', 'carrot'] is correctly sorted.
+------------------------------------------------------------------------------------------------ */
+
+const alphabetize = (arr) => {
+ arr.sort();
+ return arr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named sortByLength that takes in an array of strings and returns the same array, with the strings sorted by their length, lowest to highest.
+------------------------------------------------------------------------------------------------ */
+
+const sortByLength = (arr) => {
+ arr.sort(function(a, b){return a.length - b.length});
+ return arr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named alphabetizeBetter that takes in an array of strings and returns the same array, with the strings sorted alphabetically. Capitalization should not change the sort order of two strings.
+
+For example, ['Alphabet', 'alphabet', 'carrot', 'Zebra'] is correctly sorted, and so is ['alphabet', 'Alphabet', 'carrot', 'Zebra'].
+------------------------------------------------------------------------------------------------ */
+
+const alphabetizeBetter = (arr) => {
+ arr.sort(function(a, b){return a.toLowerCase().localeCompare(b.toLowerCase());

```

Comment:
What does .localeCompare()  do? Are you able to step through this code and explain how every part of it works?